pkg/dbtoken/api.go

```diff
@@ -3,7 +3,22 @@ package dbtoken
 // Copyright (c) Microsoft Corporation.
 // Licensed under the Apache License 2.0.
 
-const Resource = "https://dbtoken.aro.azure.com/"
+import (
+	"fmt"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// Resource returns the dbtoken domain for the provided cloud name
+func Resource(cloudName string) (string, error) {
+	switch cloudName {
+	case azure.PublicCloud.Name:
+		return "https://dbtoken.aro.azure.com/", nil
+	case azure.USGovernmentCloud.Name:
+		return "https://dbtoken.aro.azure.us/", nil
+	}
+	return "", fmt.Errorf("unsupported cloud environment")
+}
```

Comment:
I'm not a fan of this function at all. Would much rather add the whole domain in https://github.com/Azure/ARO-RP/pull/1550